Broken TypeScript declarations since 5.4.0

**Describe the bug**
Starting with 5.4.0 (no issues with 5.3.0) there are TypeScript errors when trying to build project that uses `starknet` using `tsc`

```
node_modules/starknet/dist/index.d.ts:1188:5 - error TS1038: A 'declare' modifier cannot be used in an already ambient context.

1188     declare enum TransactionType {
         ~~~~~~~

node_modules/starknet/dist/index.d.ts:2006:39 - error TS2708: Cannot use namespace 'Sequencer' as a value.

2006 declare const index_Sequencer: typeof Sequencer;
                                           ~~~~~~~~~

node_modules/starknet/dist/index.d.ts:3265:34 - error TS2304: Cannot find name 'JavaScriptValue'.

3265 declare const stringify: (value: JavaScriptValue, replacer?: any, space?: string | number | undefined, numberStringifiers?: NumberStringifier[] | undefined) => string;
                                      ~~~~~~~~~~~~~~~

node_modules/starknet/dist/index.d.ts:3265:125 - error TS2304: Cannot find name 'NumberStringifier'.

3265 declare const stringify: (value: JavaScriptValue, replacer?: any, space?: string | number | undefined, numberStringifiers?: NumberStringifier[] | undefined) => string;
                                                                                                                                 ~~~~~~~~~~~~~~~~~

node_modules/starknet/dist/index.d.ts:3267:45 - error TS2304: Cannot find name 'JavaScriptValue'.

3267 declare const stringifyAlwaysAsBig: (value: JavaScriptValue, replacer?: any, space?: string | number | undefined, numberStringifiers?: NumberStringifier[] | undefined) => string;
                                                 ~~~~~~~~~~~~~~~

node_modules/starknet/dist/index.d.ts:3267:136 - error TS2304: Cannot find name 'NumberStringifier'.

3267 declare const stringifyAlwaysAsBig: (value: JavaScriptValue, replacer?: any, space?: string | number | undefined, numberStringifiers?: NumberStringifier[] | undefined) => string;
```



**To Reproduce**
1. Clone reproducible demo from: https://github.com/Sekhmet/starknet-testnet
2. `yarn && yarn build`.
3. Building fails due to errors.

**Expected behavior**
Building with `tsc` should work.

**Desktop (please complete the following information):**

- Node version: `18.16.0`
- Starknet.js version: `5.6.0`
- TypeScript version: `5.1.6`

**Additional context**
Starting with 5.4.0 only errors related to `json` helpers are present, in later versions `declare enum TransactionType` and `typeof Sequencer` errors show up.

--- a/src/types/api/rpc.ts
+++ b/src/types/api/rpc.ts
@@ -1,66 +1,64 @@
 import { ADDRESS, CONTRACT_STORAGE_DIFF_ITEM, FELT, OPENRPC } from './openrpc';
 
-export namespace RPC {
-  export type Response = {
-    id: number;
-    jsonrpc: string;
-    result?: any;
-    error?: {
-      code: string;
-      message: string;
-    };
+export type Response = {
+  id: number;
+  jsonrpc: string;
+  result?: any;
+  error?: {
+    code: string;
+    message: string;
   };
+};
 
-  export type ChainId = OPENRPC.CHAIN_ID;
-  export type CallResponse = OPENRPC.CallResponse;
-  export type ContractAddress = ADDRESS;
-  export type Felt = FELT;
-  export type Nonce = OPENRPC.Nonce;
-  export type ContractClass = OPENRPC.ContractClass | OPENRPC.DeprecatedContractClass;
-  export type StateUpdate = OPENRPC.StateUpdate;
-  export type Transaction = OPENRPC.Transaction;
-  export type PendingTransactions = OPENRPC.PendingTransactions;
-  export type TransactionHash = OPENRPC.TransactionHash;
-  export type Trace = OPENRPC.Trace;
-  export type Traces = OPENRPC.Traces;
-  export type BlockHash = OPENRPC.BlockHash;
-  export type BlockHashAndNumber = OPENRPC.BlockHashAndNumber;
-  export type EstimateFeeResponse = OPENRPC.EstimatedFee;
-  export type GetBlockWithTxHashesResponse = OPENRPC.BlockWithTxHashes;
-  export type GetBlockWithTxs = OPENRPC.BlockWithTxs;
-  export type GetStorageAtResponse = OPENRPC.Storage;
-  export type TransactionReceipt = OPENRPC.TransactionReceipt;
-  export type GetTransactionByHashResponse = OPENRPC.Transaction;
-  export type GetTransactionByBlockIdAndIndex = OPENRPC.Transaction;
-  export type GetTransactionCountResponse = number;
-  export type GetBlockNumberResponse = OPENRPC.BlockNumber;
-  export type GetSyncingStatsResponse = OPENRPC.SyncingStatus;
-  export type EventFilter = OPENRPC.EventFilter;
-  export type GetEventsResponse = OPENRPC.Events;
-  export type InvokedTransaction = OPENRPC.InvokedTransaction;
-  export type DeclaredTransaction = OPENRPC.DeclaredTransaction;
-  export type DeployedTransaction = OPENRPC.DeployedTransaction;
-  export type SimulationFlags = OPENRPC.SimulationFlags;
-  export type BroadcastedTransaction = OPENRPC.BroadcastedTransaction;
-  export type EstimatedFee = OPENRPC.EstimatedFee;
-  export type Methods = OPENRPC.Methods;
-  export type Storage = OPENRPC.Storage;
-  export type SimulateTransactionResponse = OPENRPC.SimulatedTransactions;
+export type ChainId = OPENRPC.CHAIN_ID;
+export type CallResponse = OPENRPC.CallResponse;
+export type ContractAddress = ADDRESS;
+export type Felt = FELT;
+export type Nonce = OPENRPC.Nonce;
+export type ContractClass = OPENRPC.ContractClass | OPENRPC.DeprecatedContractClass;
+export type StateUpdate = OPENRPC.StateUpdate;
+export type Transaction = OPENRPC.Transaction;
+export type PendingTransactions = OPENRPC.PendingTransactions;
+export type TransactionHash = OPENRPC.TransactionHash;
+export type Trace = OPENRPC.Trace;
+export type Traces = OPENRPC.Traces;
+export type BlockHash = OPENRPC.BlockHash;
+export type BlockHashAndNumber = OPENRPC.BlockHashAndNumber;
+export type EstimateFeeResponse = OPENRPC.EstimatedFee;
+export type GetBlockWithTxHashesResponse = OPENRPC.BlockWithTxHashes;
+export type GetBlockWithTxs = OPENRPC.BlockWithTxs;
+export type GetStorageAtResponse = OPENRPC.Storage;
+export type TransactionReceipt = OPENRPC.TransactionReceipt;
+export type GetTransactionByHashResponse = OPENRPC.Transaction;
+export type GetTransactionByBlockIdAndIndex = OPENRPC.Transaction;
+export type GetTransactionCountResponse = number;
+export type GetBlockNumberResponse = OPENRPC.BlockNumber;
+export type GetSyncingStatsResponse = OPENRPC.SyncingStatus;
+export type EventFilter = OPENRPC.EventFilter;
+export type GetEventsResponse = OPENRPC.Events;
+export type InvokedTransaction = OPENRPC.InvokedTransaction;
+export type DeclaredTransaction = OPENRPC.DeclaredTransaction;
+export type DeployedTransaction = OPENRPC.DeployedTransaction;
+export type SimulationFlags = OPENRPC.SimulationFlags;
+export type BroadcastedTransaction = OPENRPC.BroadcastedTransaction;
+export type EstimatedFee = OPENRPC.EstimatedFee;
+export type Methods = OPENRPC.Methods;
+export type Storage = OPENRPC.Storage;
+export type SimulateTransactionResponse = OPENRPC.SimulatedTransactions;
 
-  export enum TransactionType {
-    DECLARE = 'DECLARE',
-    DEPLOY = 'DEPLOY',
-    DEPLOY_ACCOUNT = 'DEPLOY_ACCOUNT',
-    INVOKE = 'INVOKE',
-    L1_HANDLER = 'L1_HANDLER',
-  }
-
-  // Exported Diff on Sequencer (can be removed when diff resolved by new RPC v)
-  export type StorageDiffs = Array<CONTRACT_STORAGE_DIFF_ITEM>;
-  export type DeprecatedDeclaredClasses = Array<FELT>;
-  export type Nonces = Array<{
-    contract_address: ADDRESS;
-    nonce: FELT;
-  }>;
-  export type ReplacedClasses = Array<{ contract_address: ADDRESS; class_hash: FELT }>;
+export enum TransactionType {
+  DECLARE = 'DECLARE',
+  DEPLOY = 'DEPLOY',
+  DEPLOY_ACCOUNT = 'DEPLOY_ACCOUNT',
+  INVOKE = 'INVOKE',
+  L1_HANDLER = 'L1_HANDLER',
 }
+
+// Exported Diff on Sequencer (can be removed when diff resolved by new RPC v)
+export type StorageDiffs = Array<CONTRACT_STORAGE_DIFF_ITEM>;
+export type DeprecatedDeclaredClasses = Array<FELT>;
+export type Nonces = Array<{
+  contract_address: ADDRESS;
+  nonce: FELT;
+}>;
+export type ReplacedClasses = Array<{ contract_address: ADDRESS; class_hash: FELT }>;

--- a/src/types/api/sequencer.ts
+++ b/src/types/api/sequencer.ts
@@ -15,6 +15,7 @@ import {
   TransactionType,
 } from '../lib';
 
+// #region | originally not included in the namespace
 export type GetTransactionStatusResponse = {
   tx_status: TransactionStatus;
   block_hash?: string;
@@ -70,408 +71,407 @@ export type DeployedContractItem = {
 };
 
 export type SequencerIdentifier = { blockHash: string } | { blockNumber: BlockNumber };
+// #endregion
 
-export namespace Sequencer {
-  export type TransactionTraceResponse = {
-    validate_invocation?: FunctionInvocation;
-    function_invocation?: FunctionInvocation;
-    fee_transfer_invocation?: FunctionInvocation;
-    constructor_invocation?: FunctionInvocation;
-    signature: string[];
-  };
+export type TransactionTraceResponse = {
+  validate_invocation?: FunctionInvocation;
+  function_invocation?: FunctionInvocation;
+  fee_transfer_invocation?: FunctionInvocation;
+  constructor_invocation?: FunctionInvocation;
+  signature: string[];
+};
 
-  export type DeclareTransaction = {
-    type: TransactionType.DECLARE;
-    sender_address: string;
-    contract_class: ContractClass;
-    signature?: string[];
-    nonce: BigNumberish;
-    max_fee?: BigNumberish;
-    version?: BigNumberish;
-    compiled_class_hash?: string; // v2 declare
-  };
+export type DeclareTransaction = {
+  type: TransactionType.DECLARE;
+  sender_address: string;
+  contract_class: ContractClass;
+  signature?: string[];
+  nonce: BigNumberish;
+  max_fee?: BigNumberish;
+  version?: BigNumberish;
+  compiled_class_hash?: string; // v2 declare
+};
 
-  export type DeployTransaction = {
-    type: TransactionType.DEPLOY;
-    contract_definition: ContractClass;
-    contract_address_salt: BigNumberish;
-    constructor_calldata: string[];
-    nonce?: BigNumberish;
-  };
+export type DeployTransaction = {
+  type: TransactionType.DEPLOY;
+  contract_definition: ContractClass;
+  contract_address_salt: BigNumberish;
+  constructor_calldata: string[];
+  nonce?: BigNumberish;
+};
 
-  export type DeployAccountTransaction = {
-    type: TransactionType.DEPLOY_ACCOUNT;
-    class_hash: string;
-    contract_address_salt: BigNumberish;
-    constructor_calldata: string[];
-    signature?: string[];
-    max_fee?: BigNumberish;
-    version?: BigNumberish;
-    nonce?: BigNumberish;
-  };
+export type DeployAccountTransaction = {
+  type: TransactionType.DEPLOY_ACCOUNT;
+  class_hash: string;
+  contract_address_salt: BigNumberish;
+  constructor_calldata: string[];
+  signature?: string[];
+  max_fee?: BigNumberish;
+  version?: BigNumberish;
+  nonce?: BigNumberish;
+};
 
-  export type InvokeFunctionTransaction = {
-    type: TransactionType.INVOKE;
-    sender_address: string;
-    signature?: string[];
-    entry_point_type?: EntryPointType.EXTERNAL; // TODO: check this
-    calldata?: RawCalldata;
-    nonce: BigNumberish;
-    max_fee?: BigNumberish;
-    version?: BigNumberish;
-  };
+export type InvokeFunctionTransaction = {
+  type: TransactionType.INVOKE;
+  sender_address: string;
+  signature?: string[];
+  entry_point_type?: EntryPointType.EXTERNAL; // TODO: check this
+  calldata?: RawCalldata;
+  nonce: BigNumberish;
+  max_fee?: BigNumberish;
+  version?: BigNumberish;
+};
 
-  export type Transaction =
-    | DeclareTransaction
-    | DeployTransaction
-    | InvokeFunctionTransaction
-    | DeployAccountTransaction;
+export type Transaction =
+  | DeclareTransaction
+  | DeployTransaction
+  | InvokeFunctionTransaction
+  | DeployAccountTransaction;
 
-  export type AddTransactionResponse = {
-    transaction_hash: string;
-    code?: 'TRANSACTION_RECEIVED';
-    address?: string;
-    class_hash?: string;
-  };
+export type AddTransactionResponse = {
+  transaction_hash: string;
+  code?: 'TRANSACTION_RECEIVED';
+  address?: string;
+  class_hash?: string;
+};
 
-  export type GetCodeResponse = {
-    bytecode: ByteCode;
-    abi: Abi;
-  };
+export type GetCodeResponse = {
+  bytecode: ByteCode;
+  abi: Abi;
+};
 
-  export interface InvokeFunctionTransactionResponse extends InvokeFunctionTransaction {
-    transaction_hash: string;
-    entry_point_selector: string;
-  }
-
-  export type TransactionResponse =
-    | DeclareTransaction
-    | DeployTransaction
-    | InvokeFunctionTransactionResponse;
-
-  export type SuccessfulTransactionResponse = {
-    status: TransactionStatus;
-    transaction: TransactionResponse;
-    block_hash: string;
-    block_number: BlockNumber;
-    transaction_index: number;
-  };
-
-  export type FailedTransactionResponse = {
-    status: TransactionStatus.REJECTED;
-    transaction_failure_reason: {
-      code: string;
-      error_message: string;
-    };
-    transaction: TransactionResponse;
-  };
-
-  export type GetTransactionResponse = SuccessfulTransactionResponse | FailedTransactionResponse;
-
-  export type TransactionReceiptResponse =
-    | SuccessfulTransactionReceiptResponse
-    | FailedTransactionReceiptResponse;
-
-  export type SuccessfulTransactionReceiptResponse = {
-    status: TransactionStatus;
-    transaction_hash: string;
-    transaction_index: number;
-    block_hash: string;
-    block_number: BlockNumber;
-    l2_to_l1_messages: string[];
-    events: string[];
-    actual_fee: string;
-    execution_resources: ExecutionResources;
-  };
-
-  export type FailedTransactionReceiptResponse = {
-    status: TransactionStatus.REJECTED;
-    transaction_failure_reason: {
-      code: string;
-      error_message: string;
-    };
-    transaction_hash: string;
-    l2_to_l1_messages: string[];
-    events: string[];
-  };
-
-  export type GetBlockResponse = {
-    block_number: number;
-    state_root: string;
-    block_hash: string;
-    transactions: {
-      [txHash: string]: TransactionResponse;
-    };
-    timestamp: number;
-    transaction_receipts: {
-      [txHash: string]: {
-        block_hash: string;
-        transaction_hash: string;
-        l2_to_l1_messages: {
-          to_address: string;
-          payload: string[];
-          from_address: string;
-        }[];
-        block_number: BlockNumber;
-        status: TransactionStatus;
-        transaction_index: number;
-      };
-    };
-    parent_block_hash: string;
-    status: BlockStatus;
-    gas_price: string;
-    sequencer_address: string;
-    starknet_version: string;
-  };
-
-  export type CallContractTransaction = {
-    calldata?: RawCalldata;
-    max_fee?: BigNumberish;
-    version?: BigNumberish;
-    entry_point_selector: string;
-  } & (
-    | {
-        sender_address: string;
-        signature: string[];
-      }
-    | {
-        contract_address: string;
-        signature?: never;
-      }
-  );
-
-  export type CallContractResponse = {
-    result: string[];
-  };
-
-  export type InvokeEstimateFee = Omit<InvokeFunctionTransaction, 'max_fee' | 'entry_point_type'>;
-  export type DeclareEstimateFee = Omit<DeclareTransaction, 'max_fee'>;
-  export type DeployAccountEstimateFee = Omit<DeployAccountTransaction, 'max_fee'>;
-  export type DeployEstimateFee = DeployTransaction;
-
-  export type SimulateTransactionResponse = {
-    trace: TransactionTraceResponse; // diff with OPENRPC "transaction_trace"
-    fee_estimation: Sequencer.EstimateFeeResponse;
-  };
-
-  export type AccountTransactionItem =
-    | InvokeEstimateFee
-    | DeclareEstimateFee
-    | DeployEstimateFee
-    | DeployAccountEstimateFee;
-
-  /**
-   * Transaction filled with account data
-   */
-  export type AccountTransaction = AllowArray<AccountTransactionItem>;
-
-  // Support 0.9.1 changes in a backward-compatible way
-  export type EstimateFeeResponse =
-    | {
-        overall_fee: number;
-        gas_price: number;
-        gas_usage: number;
-        uint: string;
-      }
-    | {
-        amount: bigint;
-        unit: string;
-      };
-
-  export type EstimateFeeResponseBulk = AllowArray<EstimateFeeResponse>;
-
-  export type BlockTransactionTracesResponse = {
-    traces: Array<TransactionTraceResponse & { transaction_hash: string }>;
-  };
-
-  export type Storage = string;
-
-  export type StateUpdateResponse = {
-    block_hash: string;
-    new_root: string;
-    old_root: string;
-    state_diff: {
-      storage_diffs: StorageDiffs;
-      nonces: Nonces;
-      deployed_contracts: Array<DeployedContractItem>;
-      old_declared_contracts: OldDeclaredContracts;
-      declared_classes: DeclaredClasses;
-      replaced_classes: ReplacedClasses; // no definition is it array of string
-    };
-  };
-
-  export type StorageDiffs = { [address: string]: Array<StateDiffItem> };
-
-  export type StateDiffItem = { key: string; value: string };
-
-  export type Nonces = { [address: string]: Nonce };
-
-  export type Nonce = string;
-
-  export type DeployedContracts = DeployedContractItem[];
-
-  export type OldDeclaredContracts = string[];
-
-  export type DeclaredClasses = DeclaredClass[];
-
-  export type DeclaredClass = { class_hash: string; compiled_class_hash: string };
-
-  export type ReplacedClasses = string[]; // no definition is it array of string ?
-
-  export type Endpoints = {
-    get_contract_addresses: {
-      QUERY: never;
-      REQUEST: never;
-      RESPONSE: GetContractAddressesResponse;
-    };
-    add_transaction: {
-      QUERY: never;
-      REQUEST: Transaction;
-      RESPONSE: AddTransactionResponse;
-    };
-    get_transaction: {
-      QUERY: {
-        transactionHash: string;
-      };
-      REQUEST: never;
-      RESPONSE: GetTransactionResponse;
-    };
-    get_transaction_status: {
-      QUERY: {
-        transactionHash: string;
-      };
-      REQUEST: never;
-      RESPONSE: GetTransactionStatusResponse;
-    };
-    get_transaction_trace: {
-      QUERY: {
-        transactionHash: string;
-      };
-      REQUEST: never;
-      RESPONSE: TransactionTraceResponse;
-    };
-    get_transaction_receipt: {
-      QUERY: {
-        transactionHash: string;
-      };
-      REQUEST: never;
-      RESPONSE: TransactionReceiptResponse;
-    };
-    get_nonce: {
-      QUERY: {
-        contractAddress: string;
-        blockIdentifier: BlockIdentifier;
-      };
-      REQUEST: never;
-      RESPONSE: Nonce;
-    };
-    get_storage_at: {
-      QUERY: {
-        contractAddress: string;
-        key: BigNumberish;
-        blockIdentifier: BlockIdentifier;
-      };
-      REQUEST: never;
-      RESPONSE: Storage;
-    };
-    get_code: {
-      QUERY: {
-        contractAddress: string;
-        blockIdentifier: BlockIdentifier;
-      };
-      REQUEST: never;
-      RESPONSE: GetCodeResponse;
-    };
-    get_block: {
-      QUERY: {
-        blockIdentifier: BlockIdentifier;
-      };
-      REQUEST: never;
-      RESPONSE: GetBlockResponse;
-    };
-    call_contract: {
-      QUERY: {
-        blockIdentifier: BlockIdentifier;
-      };
-      REQUEST: CallContractTransaction;
-      RESPONSE: CallContractResponse;
-    };
-    estimate_fee: {
-      QUERY: {
-        blockIdentifier: BlockIdentifier;
-        skipValidate: boolean;
-      };
-      REQUEST: AccountTransactionItem;
-      RESPONSE: EstimateFeeResponse;
-    };
-    get_class_by_hash: {
-      QUERY: {
-        classHash: string;
-        blockIdentifier?: BlockIdentifier;
-      };
-      REQUEST: never;
-      RESPONSE: CompiledContract;
-    };
-    get_class_hash_at: {
-      QUERY: {
-        contractAddress: string;
-        blockIdentifier?: BlockIdentifier;
-      };
-      REQUEST: never;
-      RESPONSE: string;
-    };
-    get_state_update: {
-      QUERY: {
-        blockHash?: string;
-        blockNumber?: BlockNumber;
-      };
-      REQUEST: never;
-      RESPONSE: StateUpdateResponse;
-    };
-    get_full_contract: {
-      QUERY: {
-        contractAddress: string;
-        blockIdentifier?: BlockIdentifier;
-      };
-      REQUEST: never;
-      RESPONSE: CompiledContract;
-    };
-    estimate_message_fee: {
-      QUERY: any;
-      REQUEST: any;
-      RESPONSE: EstimateFeeResponse;
-    };
-    simulate_transaction: {
-      QUERY: {
-        blockIdentifier: BlockIdentifier;
-        skipValidate: boolean;
-      };
-      REQUEST: AccountTransaction;
-      RESPONSE: SimulateTransactionResponse;
-    };
-    estimate_fee_bulk: {
-      QUERY: {
-        blockIdentifier: BlockIdentifier;
-        skipValidate: boolean;
-      };
-      REQUEST: AccountTransaction;
-      RESPONSE: EstimateFeeResponseBulk;
-    };
-    get_block_traces: {
-      QUERY: {
-        blockHash?: string;
-        blockNumber?: BlockNumber;
-      };
-      REQUEST: never;
-      RESPONSE: BlockTransactionTracesResponse;
-    };
-    get_compiled_class_by_class_hash: {
-      QUERY: {
-        classHash: string;
-        blockIdentifier?: BlockIdentifier;
-      };
-      REQUEST: any;
-      RESPONSE: CairoAssembly;
-    };
-  };
+export interface InvokeFunctionTransactionResponse extends InvokeFunctionTransaction {
+  transaction_hash: string;
+  entry_point_selector: string;
 }
+
+export type TransactionResponse =
+  | DeclareTransaction
+  | DeployTransaction
+  | InvokeFunctionTransactionResponse;
+
+export type SuccessfulTransactionResponse = {
+  status: TransactionStatus;
+  transaction: TransactionResponse;
+  block_hash: string;
+  block_number: BlockNumber;
+  transaction_index: number;
+};
+
+export type FailedTransactionResponse = {
+  status: TransactionStatus.REJECTED;
+  transaction_failure_reason: {
+    code: string;
+    error_message: string;
+  };
+  transaction: TransactionResponse;
+};
+
+export type GetTransactionResponse = SuccessfulTransactionResponse | FailedTransactionResponse;
+
+export type TransactionReceiptResponse =
+  | SuccessfulTransactionReceiptResponse
+  | FailedTransactionReceiptResponse;
+
+export type SuccessfulTransactionReceiptResponse = {
+  status: TransactionStatus;
+  transaction_hash: string;
+  transaction_index: number;
+  block_hash: string;
+  block_number: BlockNumber;
+  l2_to_l1_messages: string[];
+  events: string[];
+  actual_fee: string;
+  execution_resources: ExecutionResources;
+};
+
+export type FailedTransactionReceiptResponse = {
+  status: TransactionStatus.REJECTED;
+  transaction_failure_reason: {
+    code: string;
+    error_message: string;
+  };
+  transaction_hash: string;
+  l2_to_l1_messages: string[];
+  events: string[];
+};
+
+export type GetBlockResponse = {
+  block_number: number;
+  state_root: string;
+  block_hash: string;
+  transactions: {
+    [txHash: string]: TransactionResponse;
+  };
+  timestamp: number;
+  transaction_receipts: {
+    [txHash: string]: {
+      block_hash: string;
+      transaction_hash: string;
+      l2_to_l1_messages: {
+        to_address: string;
+        payload: string[];
+        from_address: string;
+      }[];
+      block_number: BlockNumber;
+      status: TransactionStatus;
+      transaction_index: number;
+    };
+  };
+  parent_block_hash: string;
+  status: BlockStatus;
+  gas_price: string;
+  sequencer_address: string;
+  starknet_version: string;
+};
+
+export type CallContractTransaction = {
+  calldata?: RawCalldata;
+  max_fee?: BigNumberish;
+  version?: BigNumberish;
+  entry_point_selector: string;
+} & (
+  | {
+      sender_address: string;
+      signature: string[];
+    }
+  | {
+      contract_address: string;
+      signature?: never;
+    }
+);
+
+export type CallContractResponse = {
+  result: string[];
+};
+
+export type InvokeEstimateFee = Omit<InvokeFunctionTransaction, 'max_fee' | 'entry_point_type'>;
+export type DeclareEstimateFee = Omit<DeclareTransaction, 'max_fee'>;
+export type DeployAccountEstimateFee = Omit<DeployAccountTransaction, 'max_fee'>;
+export type DeployEstimateFee = DeployTransaction;
+
+export type SimulateTransactionResponse = {
+  trace: TransactionTraceResponse; // diff with OPENRPC "transaction_trace"
+  fee_estimation: EstimateFeeResponse;
+};
+
+export type AccountTransactionItem =
+  | InvokeEstimateFee
+  | DeclareEstimateFee
+  | DeployEstimateFee
+  | DeployAccountEstimateFee;
+
+/**
+ * Transaction filled with account data
+ */
+export type AccountTransaction = AllowArray<AccountTransactionItem>;
+
+// Support 0.9.1 changes in a backward-compatible way
+export type EstimateFeeResponse =
+  | {
+      overall_fee: number;
+      gas_price: number;
+      gas_usage: number;
+      uint: string;
+    }
+  | {
+      amount: bigint;
+      unit: string;
+    };
+
+export type EstimateFeeResponseBulk = AllowArray<EstimateFeeResponse>;
+
+export type BlockTransactionTracesResponse = {
+  traces: Array<TransactionTraceResponse & { transaction_hash: string }>;
+};
+
+export type Storage = string;
+
+export type StateUpdateResponse = {
+  block_hash: string;
+  new_root: string;
+  old_root: string;
+  state_diff: {
+    storage_diffs: StorageDiffs;
+    nonces: Nonces;
+    deployed_contracts: Array<DeployedContractItem>;
+    old_declared_contracts: OldDeclaredContracts;
+    declared_classes: DeclaredClasses;
+    replaced_classes: ReplacedClasses; // no definition is it array of string
+  };
+};
+
+export type StorageDiffs = { [address: string]: Array<StateDiffItem> };
+
+export type StateDiffItem = { key: string; value: string };
+
+export type Nonces = { [address: string]: Nonce };
+
+export type Nonce = string;
+
+export type DeployedContracts = DeployedContractItem[];
+
+export type OldDeclaredContracts = string[];
+
+export type DeclaredClasses = DeclaredClass[];
+
+export type DeclaredClass = { class_hash: string; compiled_class_hash: string };
+
+export type ReplacedClasses = string[]; // no definition is it array of string ?
+
+export type Endpoints = {
+  get_contract_addresses: {
+    QUERY: never;
+    REQUEST: never;
+    RESPONSE: GetContractAddressesResponse;
+  };
+  add_transaction: {
+    QUERY: never;
+    REQUEST: Transaction;
+    RESPONSE: AddTransactionResponse;
+  };
+  get_transaction: {
+    QUERY: {
+      transactionHash: string;
+    };
+    REQUEST: never;
+    RESPONSE: GetTransactionResponse;
+  };
+  get_transaction_status: {
+    QUERY: {
+      transactionHash: string;
+    };
+    REQUEST: never;
+    RESPONSE: GetTransactionStatusResponse;
+  };
+  get_transaction_trace: {
+    QUERY: {
+      transactionHash: string;
+    };
+    REQUEST: never;
+    RESPONSE: TransactionTraceResponse;
+  };
+  get_transaction_receipt: {
+    QUERY: {
+      transactionHash: string;
+    };
+    REQUEST: never;
+    RESPONSE: TransactionReceiptResponse;
+  };
+  get_nonce: {
+    QUERY: {
+      contractAddress: string;
+      blockIdentifier: BlockIdentifier;
+    };
+    REQUEST: never;
+    RESPONSE: Nonce;
+  };
+  get_storage_at: {
+    QUERY: {
+      contractAddress: string;
+      key: BigNumberish;
+      blockIdentifier: BlockIdentifier;
+    };
+    REQUEST: never;
+    RESPONSE: Storage;
+  };
+  get_code: {
+    QUERY: {
+      contractAddress: string;
+      blockIdentifier: BlockIdentifier;
+    };
+    REQUEST: never;
+    RESPONSE: GetCodeResponse;
+  };
+  get_block: {
+    QUERY: {
+      blockIdentifier: BlockIdentifier;
+    };
+    REQUEST: never;
+    RESPONSE: GetBlockResponse;
+  };
+  call_contract: {
+    QUERY: {
+      blockIdentifier: BlockIdentifier;
+    };
+    REQUEST: CallContractTransaction;
+    RESPONSE: CallContractResponse;
+  };
+  estimate_fee: {
+    QUERY: {
+      blockIdentifier: BlockIdentifier;
+      skipValidate: boolean;
+    };
+    REQUEST: AccountTransactionItem;
+    RESPONSE: EstimateFeeResponse;
+  };
+  get_class_by_hash: {
+    QUERY: {
+      classHash: string;
+      blockIdentifier?: BlockIdentifier;
+    };
+    REQUEST: never;
+    RESPONSE: CompiledContract;
+  };
+  get_class_hash_at: {
+    QUERY: {
+      contractAddress: string;
+      blockIdentifier?: BlockIdentifier;
+    };
+    REQUEST: never;
+    RESPONSE: string;
+  };
+  get_state_update: {
+    QUERY: {
+      blockHash?: string;
+      blockNumber?: BlockNumber;
+    };
+    REQUEST: never;
+    RESPONSE: StateUpdateResponse;
+  };
+  get_full_contract: {
+    QUERY: {
+      contractAddress: string;
+      blockIdentifier?: BlockIdentifier;
+    };
+    REQUEST: never;
+    RESPONSE: CompiledContract;
+  };
+  estimate_message_fee: {
+    QUERY: any;
+    REQUEST: any;
+    RESPONSE: EstimateFeeResponse;
+  };
+  simulate_transaction: {
+    QUERY: {
+      blockIdentifier: BlockIdentifier;
+      skipValidate: boolean;
+    };
+    REQUEST: AccountTransaction;
+    RESPONSE: SimulateTransactionResponse;
+  };
+  estimate_fee_bulk: {
+    QUERY: {
+      blockIdentifier: BlockIdentifier;
+      skipValidate: boolean;
+    };
+    REQUEST: AccountTransaction;
+    RESPONSE: EstimateFeeResponseBulk;
+  };
+  get_block_traces: {
+    QUERY: {
+      blockHash?: string;
+      blockNumber?: BlockNumber;
+    };
+    REQUEST: never;
+    RESPONSE: BlockTransactionTracesResponse;
+  };
+  get_compiled_class_by_class_hash: {
+    QUERY: {
+      classHash: string;
+      blockIdentifier?: BlockIdentifier;
+    };
+    REQUEST: any;
+    RESPONSE: CairoAssembly;
+  };
+};

--- a/src/types/index.ts
+++ b/src/types/index.ts
@@ -1,9 +1,19 @@
 export * from './account';
+export * from './calldata';
 export * from './contract';
 export * from './lib';
 export * from './provider';
 export * from './signer';
 export * from './typedData';
-export * from './api/sequencer';
-export * from './api/rpc';
-export * from './calldata';
+
+export * as RPC from './api/rpc';
+export {
+  CallL1Handler,
+  DeployedContractItem,
+  ExecutionResources,
+  FunctionInvocation,
+  GetContractAddressesResponse,
+  GetTransactionStatusResponse,
+  SequencerIdentifier,
+} from './api/sequencer';
+export * as Sequencer from './api/sequencer';

--- a/src/types/provider/response.ts
+++ b/src/types/provider/response.ts
@@ -3,8 +3,8 @@
  * Intersection (sequencer response ∩ (∪ rpc responses))
  */
 
-import { RPC } from '../api/rpc';
-import { Sequencer } from '../api/sequencer';
+import * as RPC from '../api/rpc';
+import * as Sequencer from '../api/sequencer';
 import {
   AllowArray,
   BlockStatus,

--- a/src/utils/json.ts
+++ b/src/utils/json.ts
@@ -9,12 +9,18 @@ const parseIntAsNumberOrBigInt = (x: string) => {
 };
 // NOTE: the String() wrapping is used so the behaviour conforms to JSON.parse()
 // which can accept simple data types but is not represented in the default typing
-export const parse = (x: string): any => json.parse(String(x), null, parseIntAsNumberOrBigInt);
+export const parse = (x: string): any => json.parse(String(x), undefined, parseIntAsNumberOrBigInt);
 export const parseAlwaysAsBig = (x: string): any =>
-  json.parse(String(x), null, json.parseNumberAndBigInt);
+  json.parse(String(x), undefined, json.parseNumberAndBigInt);
 
 // NOTE: the not-null assertion is used so the return type conforms to JSON.stringify()
 // which can also return undefined but is not represented in the default typing
-export const stringify = (...p: Parameters<typeof json.stringify>): string => json.stringify(...p)!;
+export const stringify = (
+  value: json.JavaScriptValue,
+  replacer?: any,
+  space?: string | number | undefined,
+  numberStringifiers?: json.NumberStringifier[] | undefined
+): string => json.stringify(value, replacer, space, numberStringifiers)!;
+
 /** @deprecated equivalent to 'stringify', alias will be removed */
 export const stringifyAlwaysAsBig = stringify;

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -14,8 +14,6 @@
     "target": "ES2020" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
     "lib": [
       "ES2020",
-      "es7",
-      "es6",
       "dom"
     ] /* Specify a set of bundled library declaration files that describe the target runtime environment. */,
     // "jsx": "preserve",                                /* Specify what JSX code is generated. */
@@ -78,7 +76,7 @@
     "forceConsistentCasingInFileNames": true /* Ensure that casing is correct in imports. */,
 
     /* Type Checking */
-    "strict": true /* Enable all strict type-checking options. */,
+    "strict": true /* Enable all strict type-checking options. */
     // "noImplicitAny": true,                            /* Enable error reporting for expressions and declarations with an implied `any` type.. */
     // "strictNullChecks": true,                         /* When type checking, take into account `null` and `undefined`. */
     // "strictFunctionTypes": true,                      /* When assigning functions, check to ensure parameters and the return values are subtype-compatible. */
@@ -100,7 +98,7 @@
 
     /* Completeness */
     // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
-    "skipLibCheck": true /* Skip type checking all .d.ts files. */
+    // "skipLibCheck": true                              /* Skip type checking all .d.ts files. */
   },
   "include": ["src/**/*"],
   "exclude": ["node_modules"]

